src/main/scala/io/gatling/build/environment/EnvironmentUtils.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2011-2022 GatlingCorp (https://gatling.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.gatling.build.environment
+
+import java.io.File
+
+import scala.io.Source
+import scala.util.{ Try, Using }
+
+object EnvironmentUtils {
+
+  private val LINE_REGEX =
+    """(?xms)
+       (?:^|\A)           # start of line
+       (                  # start variable name (captured)
+         [a-zA-Z_]        # single alphabetic or underscore character
+         [a-zA-Z0-9_.-]*  # zero or more alphnumeric, underscore, period or hyphen
+       )                  # end variable name (captured)
+       (?:=)              # assignment
+       (                  # start variable value (captured)
+         [^\#\r\n]*       # variable
```

Comment:
As we do not manage comment on line anymore and we don't manage escape, we shouldn't forbid shard `#` in the value either!

What I have in mind: url with fragment

```
REDIRECT_URL=https://auth.cloud.gatling.io/login#register
```

Reply:
Good catch !

I've added a test case.